src/System.Management.Automation/engine/hostifaces/LocalConnection.cs

```diff
@@ -836,6 +836,12 @@ private void DoCloseHelper()
                     }
                 }
 
+                if (s_IPCNamedPipeServer != null)
```

Comment:
I don't think we want to dispose here ... hosted scenarios can have no runspaces but we still want to be able to connect to them.  What I had in mind was doing this on the `AppDomain.CurrentDomain.ProcessExit += CurrentDomain_ProcessExit;` event, but I don't know if this is supported on non-Windows platforms.  But if there is an event like this where the CLR calls it on exit, that would be best.

Another option is to clean up old pipe files on start up, in a separate thread from the main thread.